.github/workflows/ci.yml

```diff
@@ -0,0 +1,160 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: CI
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
+
+env:
+  RUNTIME_BUILD_DIR: build-runtime
+
+jobs:
+  build_runtime:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e  # v2
+        with:
+          submodules: true
+      - name: "Building runtime"
+        run: |
+          ./build_tools/github_actions/docker_run.sh \
+            gcr.io/iree-oss/base@sha256:9d742e01507c292def852cbfebfae71412cff94df0ab2619f61f9a5a2a98f651 \
+            ./build_tools/cmake/build_runtime.sh \
+            "${RUNTIME_BUILD_DIR}"
+      # Using a tar archive is necessary to preserve file permissions. See
+      # https://github.com/actions/upload-artifact#maintaining-file-permissions-and-case-sensitive-files
+      # The upload action already does its own gzip compression, so it's likely
+      # unnecessary (and perhaps harmful) to do our own, though that hasn't been
+      # investigated.
+      - name: "Create build dir archive"
+        run: tar -cf ${RUNTIME_BUILD_DIR}.tar ${RUNTIME_BUILD_DIR}
+      - uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8  # v3.1.0
+        with:
+          name: "${{ env.RUNTIME_BUILD_DIR }}.tar"
+          path: "${{ env.RUNTIME_BUILD_DIR }}.tar"
+
+  test_runtime:
```

Comment:
What is the reason we don't want to use workflow_call to run these jobs?

Reply:
Because there's no need to and it creates extra files and extra code and extra nesting in the UI. With the host tools, we'll want to have a similar job that builds them without asserts, so I left that out as it's legitimately parameterized, but I don't think the extra files help for these cases. Do you disagree? I don't have *super* strong opinions

Reply:
Oh, I'm just thinking if this file will become super long after we add more jobs. We can definitely refactor it if that happens.

Reply:
Yeah it might be. I'm not sure what the right level of factoring is, since we can only trigger per-file. It's really just a cumbersome DSL to work in